common/persistence/dataInterfaces.go

```diff
@@ -800,6 +800,16 @@ type (
 		Execution workflow.WorkflowExecution
 	}
 
+	// DeleteWorkflowExecutionPartialHistoryRequest is used to delete partial workflow execution history
+	DeleteWorkflowExecutionPartialHistoryRequest struct {
+		DomainID  string
+		Execution workflow.WorkflowExecution
+		// Delete the history events from FirstEventID. Inclusive.
+		StartEventID int64
+		// Delete the history events upto NextEventID.  Not Inclusive.
+		EndEventID int64
```

Comment:
Lets not allow the user to pass upper bound for delete. Current use case is to delete the entire tail, so we should only allow callers to pass in the StartEventID.

Reply:
sure